ci/scripts/min-deps.py

```diff
@@ -62,7 +62,7 @@ def extract_min_deps(
             for extra in req.extras:
                 extra_deps = pyproject["project"]["optional-dependencies"][extra]
                 dependencies += map(Requirement, extra_deps)
-        else:
+        elif req.name != "xarray":
```

Comment:
I know this is just a draft, but if this got in, it would be a heinously hard-to-find hack. let’s not special-case dependencies in the middle of the logic here lol

Reply:
That's a totally fair point.  What should be done then?  `xarray` requires pandas 2.0 which breaks our min-deps resolution.

Reply:
And it's not relevant for the `min-deps` testing anyway.

Reply:
the exact same way as in scanpy:

```toml
[project.optional-dependencies]
test = [...]  # as before this PR
test-full = ["anndata[test]", "anndata[xarray]"]
xarray = ["xarray>=..."]
```

then modify the CI setup so it has a `TEST_EXTRA` variable defaulting to `test-full` that’s overridden for the min-deps job.

Reply:
cool!